### PR TITLE
fix(automate): redirect github auth flow to workspace if possible

### PR DESCRIPTION
**Pull request by cdriesler**
## Description & motivation

- The GitHub app auth flow currently redirects to `/functions` in all cases
- This is jarring (and, right now, incorrect) because it subtly rips people out of the "make function in workspace flow"

## Changes:

- Stores `workspaceSlug` in request session if available
- Redirects to `/workspaces/:slug/functions` at end if value is available (preserving the private function creation flow)

Validated with a personal github app locally, may be difficult to test (can help)

### CODE REVIEW

File: packages/frontend-2/components/automate/function/CreateDialog.vue [hunk 1]
```diff
@@ -77,7 +80,7 @@ const props = defineProps<{
   isAuthorized: boolean
   templates: CreatableFunctionTemplate[]
   githubOrgs: string[]
-  workspaceId?: string
+  workspace?: Pick<Workspace, 'id' | 'slug'>
```
[the remark this answers is not in the document]
Comment (cdriesler) [reply packages/frontend-2/components/automate/function/CreateDialog.vue]: roger, done

File: packages/frontend-2/components/automate/function/EditDialog.vue [hunk 2]
```diff
@@ -18,12 +18,20 @@ import { difference, differenceBy } from 'lodash-es'
 import { useForm } from 'vee-validate'
 import { useUpdateAutomateFunction } from '~/lib/automate/composables/management'
 import type { FunctionDetailsFormValues } from '~/lib/automate/helpers/functions'
-import type { Workspace } from '~/lib/common/generated/gql/graphql'
+import { graphql } from '~/lib/common/generated/gql'
+import type { AutomateFunctionEditDialog_WorkspaceFragment } from '~/lib/common/generated/gql/graphql'
+
+graphql(`
+  fragment AutomateFunctionEditDialog_Workspace on Workspace {
```
Comment (fabis94) [inline packages/frontend-2/components/automate/function/EditDialog.vue:25]: I don't see this one used in other fragments or queries

**Review by fabis94 — APPROVED** [review]
nearly there

File: packages/frontend-2/components/automate/function/EditDialog.vue (hunk 2 above)
Comment (cdriesler) [reply packages/frontend-2/components/automate/function/EditDialog.vue:25]: updated 👍

**Pull request merged**